### DISCUSSION TITLE
Display short network address

### DISCUSSION THREAD

**Discussion by grafalex82** [Koenkk/zigbee2mqtt#7603] [General]
I am developing my zigbee device, and seeing a correct short network address is extremely valuable to me. Unfortunately I see some discrepancies when Z2M display the network address. 

Here is a wireshark screenshot of a random device on my network
![image](https://user-images.githubusercontent.com/23584882/119561782-471d6500-bdae-11eb-8684-429b40cf92be.png)
The short address is 0xe6ad

Here is a screenshot of a zigbee map in home assistant showng the same device
![image](https://user-images.githubusercontent.com/23584882/119561914-6d430500-bdae-11eb-8a20-f104832309d3.png)
On this screenshot the network address is correct (59053), but having it in a decimal form instead of hexadecimal is very inconvenient for debugging. 

**Question 1:** Would it be possible to show short network address on the map in a heaxadecimal form? I even localized the place (networkMap.js line 86) where this should happen, but unfortunately my JS knowledge is not good enough to complete this task.

Here is a screenshot of Z2M dashboard for the same device
![image](https://user-images.githubusercontent.com/23584882/119562123-a8453880-bdae-11eb-8d80-5216c8892290.png)
For some reason network address is shown here as 0xd80b instead of 0xe6ad

**Question 2** Is this a bug or feature, that network address is different from one really used in the network?

**Comment by Koenkk** [comment]
Question 1: please create a feature request here: https://github.com/nurikk/z2m-frontend/issues
Question 2: could be fixed after a restart of z2m, network addresses can change.

**Reply by grafalex82** [reply]
On Question 1 - submitted https://github.com/nurikk/z2m-frontend/issues/555 as you requested

On Question 2. Indeed, restart Z2M helped. But why this happen? I saw in the wireshark that network addresses can change over time, but rejoin response message has the new address. Can Z2M handle these messages somehow?

**Reply by sjorge** [reply]
> Question 1: please create a feature request here: https://github.com/nurikk/z2m-frontend/issues

Isn't the code that generated the raw or graphviz code in z2m, instead of the frontend?

**Reply by grafalex82** [reply]
I guess this is graphviz code

**Reply by Koenkk** [reply]
Q1: z2m just provide the network address as a number, it is up to the frontend to visualize it.
Q2: currently z2m doesn't republish the devices list when the network address changes (but it definitely does update the network address internally, otherwise you would not be able to control it anymore)

**Reply by OleksandrMasliuchenkoGL** [reply]
Q1: You are right, frontend should be responsible for visualizing. But in this case Z2M is building a string consisting of a full 64bit network address, and short network address in parenthesis. And the short address is in dec, not in hex.
e.g. "0x123456789abcdef (12345)"

Q2: Is it possible to get/display actual information somehow?

**Reply by Koenkk** [reply]
Q1: ah, I thought you meant the z2m frontend, but the frontend shown in HA is the graphviz one (which is created directly by z2m). Both plantuml and graphviz maps will show the network address as hex now.

Q2: fixed

Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/how_tos/how-to-switch-to-dev-branch.html)

**Reply by grafalex82** [reply]
Thank you very much for your effort. I'll switch to a dev branch for testing.

**Reply by OleksandrMasliuchenkoGL** [reply]
As for Q1. Device addresses in node boxes is now in hex. Thank you!
But I noticed that there are route links that reference addresses in dec.

![image](https://user-images.githubusercontent.com/61513097/120173377-a5ae6d00-c20c-11eb-94a8-9c74edc69ccb.png)

Would you be able to convert to hex it as well, please?

**Reply by Koenkk** [reply]
@OleksandrMasliuchenkoGL fixed, 
Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/how_tos/how-to-switch-to-dev-branch.html)

**Reply by grafalex82** [reply]
I confirm the fix. Thank you again for this change, this is really appreciated.
